tests/integration/test_pagure.py

```diff
@@ -19,54 +19,79 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
-
 import os
 import subprocess
+from datetime import datetime
 from pathlib import Path
 
-import pytest
 from ogr.services.pagure import PagureService
+from vcr import use_cassette
 
 from tests.spellbook import git_set_user_email
 
 
-@pytest.mark.skipif(
-    condition=True, reason="Don't interact with a real pagure instance by default"
-)
+RESPONSE_HEADERS_TO_DROP = [
+    "X-Fedora-RequestID",
+    "content-security-policy",
+    "set-cookie",
+]
+
+
+def drop_sensitive_response_data(response):
+    for h in RESPONSE_HEADERS_TO_DROP:
+        try:
+            del response["headers"][h]
+        except KeyError:
+            pass
+    return response
+
+
 def test_basic_distgit_workflow(tmpdir):
-    pagure_token = os.getenv("PAGURE_TOKEN")
+    # if pagure_token is not set and we're recording, we'll get a token failure,
+    # but that's expected so no need to take care of it
```

Comment:
There is no way how to check the mode? It can cause misleading messages in the CI.

Reply:
the only error messages I got was that token is not valid - pretty clear

when you tamper with the call stack, you will always get misleading error messages

Reply:
It will be funny with GitHub when you randomly receive API limit errors since it works also unauthenticated.

I don't say it's a big problem, but I already need to describe this problem to various contributors and sometimes also people from the team was confused. (The warnings from API are missing important info -- you need to set a specific env. var. to fix that.)